Replicator/Checkpoint.cc

```diff
@@ -74,32 +74,44 @@ namespace litecore { namespace repl {
 
 
     void Checkpoint::readJSON(slice json) {
-        resetLocal();
-        _remote = {};
         if (json) {
             Doc root = Doc::fromJSON(json, nullptr);
-            if (!root) {
+            if (!root)
```

Comment:
This reversal of flow logic makes things hard to follow.  The former was easier:  If fail, return early.  This is "If success, return early, else log error and continue"

Reply:
It is my intention to avoid calling resetLocal() and reseting _remote twice. Agreed that it's hard to understand so it's not worth it. I will revert it back